### ISSUE TITLE
The flame graph is incorrect when using go 1.23.

### ISSUE THREAD

**Issue by manchurio** [felixge/fgprof#33]
```go
package main

import (
	"fmt"
	"github.com/felixge/fgprof"
	"net/http"
	"os"
	"time"
)

func main() {
	cpuf, err := os.Create("fgprof.out")
	if err != nil {
		return
	}
	stop := fgprof.Start(cpuf, fgprof.FormatPprof)
	defer stop()
	for range 30 {
		networkTask()
		cpuTask()
	}
}

func networkTask() {
	resp, err := http.Get("http://google.com")
	if err != nil {
		fmt.Println(err)
		return
	}
	defer resp.Body.Close()
}

func cpuTask() {
	start := time.Now()
	for time.Since(start) < 100*time.Millisecond {
		for i := 0; i < 10000; i++ {
			_ = i
		}
	}
}
```
### As shown in the image, cputask and networkTask should be at the same level, but now cpuTask is below networkTask.
![image](https://github.com/user-attachments/assets/0cbd0c0d-f21d-4c20-a447-8c6d12cfa954)

### As a comparison, this is for Go 1.22.
![image](https://github.com/user-attachments/assets/6328e534-2d17-40ea-ac9e-a79758320363)

**Comment by felixge** [comment]
Thanks for reporting this. This might be an upstream regression, potentially in some patches submitted by my colleague and me for the go1.23 release. I'll investigate.

**Comment by felixge** [comment]
I can confirm that this is a regression I introduced upstream, sorry about that.

I've [submitted a fix](https://go-review.googlesource.com/c/go/+/609815) and will ask for getting it back-ported to a go1.23.x release.

Meanwhile I've created a [workaround](https://github.com/felixge/fgprof/commit/3b04a2333b6d470171c1f244ce1ec47882c8e5bc) and tagged a new `v0.9.5` release that contains it. Upgrading should fix the issue on your end.
